File: INE5416-Paradigmas-de-programacao/Atividade-2/1025-marmore/main.py
case = 1
while True:

    n, m = map(int, input().split())

    if (n == 0 and m == 0): break

    nums = []

    for i in range (n):
        nums.append(int(input()))

    nums.sort()

    print("CASE# %d:" % case)
    for i in range (m):
        found = False
        a = int(input())
        size = len(nums)
        pos = size//2
        stop = 0
        ajuste = size//4
        while stop < 3:
            if nums[pos] == a:
                while (pos > 0 and nums[pos-1] == a):
                    pos -= 1
                print("%d found at %d" % (a, pos+1))
                found = True
                break
            elif nums[pos] < a:
                pos = pos + ajuste
                if pos >= len(nums):
                    pos = len(nums) - 1
            else:
                pos = pos - ajuste
                if pos < 0:
                    pos = 0
            if ajuste == 1:
                stop += 1
            ajuste = ajuste % 2 == 0 and ajuste//2 or (ajuste//2) + 1
        if not found:
            print("%d not found" % a)
    
    case += 1

File: INE5416-Paradigmas-de-programacao/Atividade-2/1025-marmore/main.cpp
#include <stdlib.h>
#include <stdio.h>
#include <vector>
#include <algorithm>

using namespace std;

int main(int argc, char const *argv[])
{
    int caso = 1;
    while (true) {
        int n, m;
        scanf("%d %d", &n, &m);

        if (n == 0 and m == 0) break;

        vector<int> nums = vector<int>();

        for (int i = 0; i < n; i++) {
            int a;
            scanf("%d", &a);
            nums.push_back(a);
        }

        sort(nums.begin(), nums.end());

        printf("CASE# %d:\n", caso);
        for (int i = 0; i < m; i++) {
            bool found = false;
            int a;
            scanf("%d", &a);
            int size = nums.size();
            int pos = size / 2;
            int stop = 0;
            int ajuste = size / 4;
            while (stop < 3) {
                if (nums[pos] == a) {
                    while (pos > 0 && nums[pos-1] == a) {
                        pos--;
                    }
                    printf("%d found at %d\n", a, pos+1);
                    found = true;
                    break;
                } else if (nums[pos] < a) {
                    pos = pos + ajuste;
                    if (pos >= nums.size()) pos = nums.size() -1;
                } else {
                    pos = pos - ajuste;
                    if (pos < 0) pos = 0;
                }

                if (ajuste == 1) stop++;

                ajuste = ajuste % 2 == 0 ? ajuste / 2 : ajuste / 2 + 1;
            }
            if (!found) printf("%d not found\n", a);
        }
        caso++;
    }
    return 0;
}
